Automattic/woocommerce-payments
# Remove leftovers of the "Boost your sales by accepting new payment methods" task

Fixes #3434

#### Changes proposed in this Pull Request

Remove an unused component and variables.

* Displaying the wizard to set up the UPE payment methods at `&path=%2Fpayments%2Fadditional-payment-methods`, even when UPE preview is disabled. There shouldn't be links or redirects taking the user to this page when the preview is disabled
* Deleting the `MethodsSelector` component that used to be displayed on this page when UPE preview was disabled
* Deleting the tests and variables that were only used by this component or by the removed task functionality

#### Testing instructions

Functional instructions:
* Make sure UPE is disabled. Disable it via WCPay Dev tools. If you do it via WCPay settings the admin note might not show up
* Make sure the UPE preview is enabled. It's enabled by default, but in case it's not, simply remove `wcpay_additional_methods_setup_completed` from wp_options
* Go to Payments -> Overview
* Find the admin note titled "Boost your sales by accepting new payment methods" and click on "Enable on your store"
* It should take you to the additional-payment-methods wizard at `&path=%2Fpayments%2Fadditional-payment-methods` and everything should continue to work as expected there
* Disable the UPE preview by setting `_wcpay_feature_upe_settings_preview` to '0' in wp_options. Looks like the notes are cached. Deactivating WCPay and activating it back prevents the note from persisting
* There shouldn't be links or redirects going to the additional-payment-methods wizard (it's probably simpler to check this in the code)
* If you go to `/admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods` directly, the wizard should still be there and it shouldn't error when you go through it

In the code:
* The variable `wcpaySettings.additionalMethodsSetup.isTaskVisible` that's localized to the JS files shouldn't exist or be used
* The variable `wcpaySettings.additionalMethodsSetup.isSetupCompleted` that's localized to the JS files shouldn't exist or be used
* The wp_option `wcpay_additional_methods_setup_completed` shouldn't be used anywhere
* The component `MethodsSelector` that used to be defined at `client/additional-methods-setup/methods-selector/index.js` shouldn't exist or be used, along with the code used only by it
* Ideally, there shouldn't be any code that's left unused as the result of the removal of the "Boost your sales by accepting new payment methods" task at https://github.com/Automattic/woocommerce-payments/pull/3134

-------------------


**Post merge**
